techniques/system/distributePolicy/1.0/rsyslogConf.st

```diff
@@ -46,77 +62,61 @@ bundle agent install_rsyslogd {
-              comment => "Removing the logging of all in the database";  
-
-  packages:
-
-    policy_server.!SuSE.!redhat.!reports_disabled.!role_rudder_relay_promises_only::
-      "rsyslog"
-        package_policy  => "add",
-        package_method  => generic,
-        classes => cf2_if_else("rsyslog_installed", "cant_install_rsyslog"),
-        comment => "Installing rsyslog using apt backports";
-
-      "rsyslog-pgsql"
-        package_policy  => "add",
-        package_method  => generic,
-        classes => cf2_if_else("rsyslog_pgsql_installed", "cant_install_rsyslog_pgsql"),
-        comment => "Installing rsyslog_pgsql using apt backports";
-
-    policy_server.!SuSE.redhat.!reports_disabled.!role_rudder_relay_promises_only::
-      "rsyslog"
-        package_policy  => "add",
-        package_method  => rudder_yum,
-        classes => cf2_if_else("rsyslog_installed", "cant_install_rsyslog"),
-        comment => "Installing rsyslog using apt backports";
-
-      "rsyslog-pgsql"
-        package_policy  => "add",
-        package_method  => rudder_yum,
-        classes => cf2_if_else("rsyslog_pgsql_installed", "cant_install_rsyslog_pgsql"),
-        comment => "Installing rsyslog_pgsql using apt backports";
+        classes => classes_generic("rudder_rsyslog_pgsql"),
+        comment => "Removing the logging of all in the database";  
+
+  commands:
+    policy_server.!SuSE.(rudder_rsyslog_conf_repaired|rsyslog_inc_repaired|rudder_rsyslog_historical_conf_repaired|rudder_rsyslog_pgsql_repaired)::
+      "${paths.path[service]} rsyslog"
+        args => "restart",
+        classes => classes_generic("rsyslog_restarted"),
+        comment => "restarting rsyslog";
+    policy_server.SuSE.(rudder_rsyslog_conf_repaired|rsyslog_inc_repaired|rudder_rsyslog_historical_conf_repaired|rudder_rsyslog_pgsql_repaired)::
+      "${paths.path[service]} syslog"
+        args => "restart",
+        classes => classes_generic("rsyslog_restarted"),
+        comment => "restarting rsyslog";
 
-  reports:
 
-    cant_install_rsyslog|cant_install_rsyslog_pgsql::
-      "Fatal : Can't install rsyslog or rsyslog_pgsql on the Rudder root server !";
+  reports:
 
-    cannot_copy_rudder_rsyslog_conf::
+    rudder_rsyslog_conf_not_ok::
       "Fatal : Can't copy the rsyslog configuration !";
 
-    rsyslog_inc_failed::
+    rsyslog_inc_not_ok::
       "Fatal : Can't enable the rsyslog include directory !";
 
-    cannot_update_rudder_rsyslog_pgsql::
+    rudder_rsyslog_pgsql_not_ok::
       "Fatal : Cannot update the pgsql configuration !";
 
-    service_restart_rsyslog_repaired|service_restart_syslog_repaired::
+    rsyslog_restarted_repaired::
       "Info : Restarted rsyslog";
 
-    service_restart_rsyslog_not_ok|service_restart_syslog_not_ok::
+    rsyslog_restarted_not_ok::
       "Fatal : Can't restart rsyslog !";
 
+    rsyslog_port_too_many::
+      "Warning: rsyslog port number has been set several times in /opt/rudder/etc/rudder-web.properties";
```

Comment:
Since this last report can only happen if in initial promises, you could surround it with an "&if(INITIAL)&" as the section the rsyslog_port_too_many class is defined in above.